### ISSUE TITLE
web_platform: Canvas readback after device lost/destroyed

### ISSUE THREAD

**Issue by kainino0x** [gpuweb/cts#3252]
I believe that according to the WebGPU spec, canvas [readback (via drawImage etc.)](https://gpuweb.github.io/gpuweb/#abstract-opdef-get-a-copy-of-the-image-contents-of-a-context) should interact with device loss (and destroy()) in the same way I described in #3251:

> * After the device is lost but before calling getCurrentTexture, it should keep returning the last displayed frame
> * After the device is lost and after calling getCurrentTexture, it should return:
>   
>   * a solid black image, if configure()d as opaque
>   * a blank transparent image, if configure()d as transparent

Rough test cases:
- Render something, destroy the device, check readback, wait a few frames, check readback again
- Render something, destroy the device, wait a few frames, call getCurrentTexture, check readback, wait a few frames, check readback again

**Comment by kainino0x** [comment]
@perryuwang
This one is a lot more involved than the reftest as there are a lot of different ways to read back from a canvas, but this is the file where these tests will be added:
https://github.com/gpuweb/cts/blob/873305ab6a2eae37dc899856ccd55bebe8e94f8a/src/webgpu/web_platform/canvas/readbackFromWebGPUCanvas.spec.ts
If you are interested in picking this up, see the intro documentation on how to develop on these locally.

Basically I think we'll add a "parameter" to most of these tests for whether the device should be destroyed before the readback, which shouldn't change the readback result.
For the readback after destroy AND getCurrentTexture, I'm not sure how to add those to the existing test cases.
